src/internal/utils.ts

```diff
@@ -11,9 +13,23 @@ export function getAndValidateString(parsedArgs: minimist.ParsedArgs, option: st
   return value;
 }
 
-export function getNetwork(chainId: number): Network {
+export async function getNetwork(chainId: number, networkClient?: NetworkClient): Promise<string> {
   const network = fromChainId(chainId);
   if (network === undefined) {
```

Comment:
I see an actual network has priority over a private one, but I couldn't find any documentation in the Defender docs about the `chainId` of a private/forked network. Are we considering the case when a new network is added to Defender and it happens to use the same `chainId` as multiple private networks already created in Defender?

Reply:
Updated this to handle duplicate `chainId` and support a `DEFENDER_NETWORK` env var to disambiguate.